cmd/githubCheckBranchProtection.go

```diff
@@ -0,0 +1,64 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/google/go-github/v32/github"
+
+	"github.com/pkg/errors"
+
+	piperGithub "github.com/SAP/jenkins-library/pkg/github"
+)
+
+type githubRepositoriesService interface {
+	GetBranchProtection(ctx context.Context, owner, repo, branch string) (*github.Protection, *github.Response, error)
+}
+
+func githubCheckBranchProtection(config githubCheckBranchProtectionOptions, telemetryData *telemetry.CustomData) {
+	ctx, client, err := piperGithub.NewClient(config.Token, config.APIURL, "")
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Failed to get GitHub client")
+	}
+
+	err = runGithubCheckBranchProtection(ctx, &config, telemetryData, client.Repositories)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("GitHub branch protection check failed")
+	}
+}
+
+func runGithubCheckBranchProtection(ctx context.Context, config *githubCheckBranchProtectionOptions, telemetryData *telemetry.CustomData, ghRepositoriesService githubRepositoriesService) error {
+	ghProtection, _, err := ghRepositoriesService.GetBranchProtection(ctx, config.Owner, config.Repository, config.Branch)
+	if err != nil {
+		return errors.Wrap(err, "failed to read branch protection information")
+	}
+
+	// validate required status checks
+	for _, check := range config.RequiredChecks {
+		var found bool
+		for _, context := range ghProtection.GetRequiredStatusChecks().Contexts {
+			if check == context {
+				found = true
+			}
+		}
+		if !found {
+			return fmt.Errorf("required status check '%v' not found in branch protection configuration", check)
```

Comment:
Could it be helpful here to log the found checks (contexts)?

Reply:
good point, will add this ...